pkg/frontend/querymiddleware/astmapper/sharding.go

```diff
@@ -167,14 +168,14 @@ func (summer *shardSummer) shardAndSquashFuncCall(expr *parser.Call, stats *Mapp
 		for i, arg := range clonedCall.Args {
 			switch typedArg := arg.(type) {
 			case *parser.SubqueryExpr:
-				subExprMapped, err := subSummer.Map(typedArg.Expr, nil)
```

Comment:
Uhm... we didn't pass `stats` here. Was a bug?

Reply:
I checked the code and I think this was unintentional. I think we never noticed because in practice this call shouldn't lead to any call to `AddShardedQueries()`.

Reply:
I also noticed that 👀  I'm not sure if it is a bug, or if it was forgotten. But the tests still pass, so could it be that we should add some tests for this specific case?